doc/source/optimization.md

```diff
@@ -1,67 +1,264 @@
+of pods that isn't allowed to be disrupted. Pods that are not allowed to be
+disrupted are for example real user pods, various system pods, and some
+JupyterHub pods (without a permissive
+[PodDisruptionBudget](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/)).
+Consider for example that you add a lot of users during the daytime. New nodes
+are added by the CA. Some system pod ends up on the new nodes along with the
+user pods for some reason. At night when the
+[*culler*](user-management.html#culling-user-pods) has removed many inactive
+pods by some nodes was free from users but there was still a single system pod
+stopping the CA from removing it.
+
+To avoid these scale down failures, we recommend using a *dedicated node pool*
+for the user pods. To accomplish this, we can use [*taints and
+tolerations*](https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/).
+If we add a taint to all the nodes in the node pool, and a toleration on the
+user pods to tolerate being scheduled on a tainted node, we have practically
+dedicated the node pool to be used by user pods.
+
+#### Using a user dedicated node pool
+
+To make users schedule on a dedicated node for them, you need to do the following:
+
+1. Setup a node pool (with autoscaling), a certain label, and a certain taint.
+
+    How you do this will depend on the cloud provider, also note that cloud
+    providers often have their own labels separate from kubernetes labels. This
+    needs to be a kubernetes label.
+
+    ```
+    # label=value
+    hub.jupyter.org/node-purpose=user
+    
+    # taint=value:effect (you may need to replace '/' with '_')
+    hub.jupyter.org/dedicated=user:NoSchedule
+    ```    
+
+2. Make user pods require to be scheduled on the node pool setup above
+
+    If you don't require the user pods to schedule on their dedicated node, you
+    may fill up the nodes where the other software runs. This can cause a `helm
+    upgrade` command to fail because you may have run out of resources for
+    non-user pods that cannot schedule on the autoscaling node pool as they need
+    during a rolling update for example.
+
+    The default setting is to make user pods *prefer* schedule on nodes with the
+    `hub.jupyter.org/node-purpose=user` label, but you can also make it
+    *require* to do so using the configuration below.
+
+    ```yaml
+    scheduling:
+      userPods:
+        nodeAffinity:
+          matchNodePurpose: require
+    ```
+
+NOTE: If you end up not using a dedicated node pool for users and want to scale
+down efficiently, you will need to learn about PodDisruptionBudget and to quite
+a bit more of work to avoid ending up with almost empty nodes not scaling down.
+
+#### Using resources efficiently (the user scheduler)
+
+If you have users coming online and others being culled by inactivity, but on
+average you are needing less and less nodes. How will you free up a node so it
+can be scaled down?
```

Comment:
```
If you have users starting new servers while the total number of active users declining,
how will you free up a node so it can be scaled down?
```